tests/integration/submission/test_upload.py

```diff
@@ -447,47 +527,99 @@ def test_dont_enforce_file_hash_size_uniqueness(
 
-    for file_state in ALLOWED_STATES:
-        # set node's url state in indexd
-        indexd_doc = indexd_client.get(did)
-        set_indexd_state(indexd_doc, url, file_state)
+    # Actual update call - should fail
+    resp = client.put(BLGSP_PATH, headers=submitter,
+                      data=json.dumps(new_version))
+    doc_after_put = indexd_client.get(node_id)
 
-        # check if updatable
-        assert entity.is_updatable_file_node(node)
+    assert resp.status_code == 400
+    assert doc_original.to_json() == doc_after_put.to_json()
 
-    for file_state in DISALLOWED_STATES:
-        # set node's url state in indexd
-        indexd_doc = indexd_client.get(did)
-        set_indexd_state(indexd_doc, url, file_state)
-        # check if not updatable
-        assert not entity.is_updatable_file_node(node)
+
+@pytest.mark.config_toggle(
+    parameters={
+        'CREATE_REPLACEABLE': True,
+    }
+)
+def test_update_multiple_one_fails(
+        client_toggled, pg_driver, submitter, cgci_blgsp, indexd_client):
+    """Test that updating multiple nodes at the same time with one of the nodes
+    being in incorrect state will result in a failure of the whole update
+    transaction, i.e. no nodes should be updated
+
+    Test performs the following:
+    1. Create several nodes + submitted unaligned reads node
+    2. Create submitted aligned reads node
+    3. Set submitted unaligned reads node state to "submitted", which will
+    prevent the node from being updated
+    4. Send a request to update both aligned and unaligned reads nodes
+    5. Verify that none of the nodes were updated
+    """
+
+    resp, sur_entity = data_file_creation(
+        client_toggled, submitter,
+        sur_filename='submitted_unaligned_reads.json')
+
+    assert resp['code'] == 201
+
+    sur_node_id = sur_entity['id']
+
+    sar_old = read_json_data(
+        os.path.join(DATA_DIR, 'submitted_aligned_reads.json'))
+
+    # Create Submitted aligned reads file node
+    resp = client_toggled.post(BLGSP_PATH, headers=submitter,
+                               data=json.dumps(sar_old))
+    assert resp.status_code == 201
+
+    sar_node_id = resp.json['entities'][0]['id']
+
+    sur_doc_old = indexd_client.get(sur_node_id).to_json()
+    sar_doc_old = indexd_client.get(sar_node_id).to_json()
+
+    # Modify submitted aligned reads metadata
+    sar_new = copy.deepcopy(sar_old)
+    sar_new['file_name'] = 'submitted_aligned_reads_new.bam'
+    sar_new['md5sum'] = 'ac4ca6d336b57a94b34e923d3d7a627a'
+    sar_new['file_size'] = 12288
+
+    # Load updated submitted unaligned reads metadata
+    sur_new = read_json_data(
+        os.path.join(DATA_DIR, 'submitted_unaligned_reads.json'))
+
+    # Set unaligned reads node state to 'submitted', which will prevent the
+    # node update
+    with pg_driver.session_scope():
+        sur_node = pg_driver.nodes().get(sur_node_id)
+        sur_node.state = 'submitted'
```

Comment:
So this should be happy if this isn't submitted? Test the happy path too (not sure if you already have)

Reply:
We have it in `sheepdog.tests.integration.submission.test_versioning::test_update_data_file_entity`, which essentially updates multiple nodes.